### ISSUE TITLE
Vagrant image gives internal server error from galaxy

### ISSUE THREAD

**Issue by dctrud** [galaxyproject/planemo#122]
After starting up a planemo vagrant VM, and going to localhost:8010 I see a galaxy internal server error.

ssh-ing into the vagrant VM and looking at the log it seems there is a kombu problem:

```python
10.0.2.2 - - [13/Apr/2015:03:21:32 +0000] "GET / HTTP/1.0" 500 - "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_2) AppleWebKit/600.4.10 (KHTML, like Gecko) Version/8.0.4 Safari/600.4.10"
Error - <class 'sqlalchemy.exc.InvalidRequestError'>: One or more mappers failed to initialize - can't proceed with initialization of other mappers.  Original exception was: When initializing mapper Mapper|Queue|kombu_queue, expression 'Message' failed to locate a name ("name 'Message' is not defined"). If this is a class name, consider adding this relationship() to the <class 'kombu.transport.sqlalchemy.Queue'> class after both dependent classes have been defined.
URL: http://localhost/
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/middleware/error.py', line 149 in __call__
  app_iter = self.application(environ, sr_checker)
File '/opt/galaxy/galaxy-app/eggs/Paste-1.7.5.1-py2.7.egg/paste/recursive.py', line 84 in __call__
  return self.application(environ, start_response)
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/middleware/remoteuser.py', line 141 in __call__
  return self.app( environ, start_response )
File '/opt/galaxy/galaxy-app/eggs/Paste-1.7.5.1-py2.7.egg/paste/httpexceptions.py', line 633 in __call__
  return self.application(environ, start_response)
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/base.py', line 133 in __call__
  return self.handle_request( environ, start_response )
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/base.py', line 160 in handle_request
  trans = self.transaction_factory( environ )
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/webapp.py', line 73 in <lambda>
  self.set_transaction_factory( lambda e: self.transaction_chooser( e, galaxy_app, session_cookie ) )
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/webapp.py', line 104 in transaction_chooser
  return GalaxyWebTransaction( environ, galaxy_app, self, session_cookie )
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/webapp.py', line 205 in __init__
  self._ensure_valid_session( session_cookie )
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/webapp.py', line 412 in _ensure_valid_session
  user_for_new_session = self.get_or_create_remote_user( remote_user_email )
File '/opt/galaxy/galaxy-app/lib/galaxy/web/framework/webapp.py', line 527 in get_or_create_remote_user
  user = self.sa_session.query( self.app.model.User
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/orm/scoping.py', line 150 in do
  return getattr(self.registry(), name)(*args, **kwargs)
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/orm/session.py', line 1165 in query
  return self._query_cls(entities, self, **kwargs)
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/orm/query.py', line 108 in __init__
  self._set_entities(entities)
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/orm/query.py', line 118 in _set_entities
  self._set_entity_selectables(self._entities)
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/orm/query.py', line 151 in _set_entity_selectables
  ent.setup_entity(*d[entity])
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/orm/query.py', line 3036 in setup_entity
  self._with_polymorphic = ext_info.with_polymorphic_mappers
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/util/langhelpers.py', line 725 in __get__
  obj.__dict__[self.__name__] = result = self.fget(obj)
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/orm/mapper.py', line 1877 in _with_polymorphic_mappers
  configure_mappers()
File '/opt/galaxy/galaxy-app/eggs/SQLAlchemy-0.9.8-py2.7-linux-x86_64-ucs4.egg/sqlalchemy/orm/mapper.py', line 2586 in configure_mappers
  raise e
InvalidRequestError: One or more mappers failed to initialize - can't proceed with initialization of other mappers.  Original exception was: When initializing mapper Mapper|Queue|kombu_queue, expression 'Message' failed to locate a name ("name 'Message' is not defined"). If this is a class name, consider adding this relationship() to the <class 'kombu.transport.sqlalchemy.Queue'> class after both dependent classes have been defined.
```

**Comment by dctrud** [comment]
Looks same as: http://dev.list.galaxyproject.org/Problems-after-upgrade-td4666934.html

Restarting galaxy process via supervisorctl in the VM got me a working galaxy after couple of goes.

**Comment by dannon** [comment]
Hey @dctrud -- yep, known galaxy issue (not specific to planemo at all).  I've been working on this recently and have a new version of the kombu egg to test that I'm hopeful will resolve the issue, so it's actually great to know that this is reproducible in the VM!

**Comment by dctrud** [comment]
Just a note that it's not very reproducible in the planemo vagrant VM across different machines. I get it frequently if I'm using Vagrant on my MacBook Air running Fedora 21 with VirtualBox etc from repos. I've not seen it despite various restarts of the VM when I'm using a CentOS 7 Intel NUC at home, or running Vagrant under Windows on my work ThinkPad.

Away from the planemo VM I've had seen the kombu issue only once in maybe 100 restarts of Galaxy on our production RHEL6.6 virtualized server.

**Comment by dannon** [comment]
Yeah, in *many* restarts of the vagrant box on my machine I'm not able to reproduce it at all.  I'm going to have not one but *two* potential fixes in today, so if you're willing to test it that'd be awesome.

**Comment by dctrud** [comment]
Absolutely - just let me know when something is ready to test and I'll be happy to do that.

**Comment by dannon** [comment]
@dctrud Ok, I tried reproducing this in docker-galaxy-stable too, no dice on my hardware, so I'm going to take you up on kicking those few ideas your way.  Two branches in my galaxy fork (off -dev) have possible fixes; the kombu update I'll probably push forward with anyway, the other one is totally a hack and questionable:
github.com/dannon/galaxy.git update_kombu
github.com/dannon/galaxy.git mapperinit_hack

**Comment by dctrud** [comment]
Hi Dannon,

So - I checked the update_kombu branch on all the machines / VM I'm using here at work and no kombu issue seen despite many start/stops of Galaxy. However, for sanity's sake I tried Galaxy with old kombu and saw no issues either....

I really need to try this on my home MacBook / vagrant as that's where the other day it was happening on perhaps 50% of startups of Galaxy. I'm afraid I didn't have time to do that at home last night.

Really frustrating - sometimes I see it often, sometimes it never seems to occur.

**Comment by dannon** [comment]
Thanks for checking it out!  I'm going to go ahead and get the kombu/amqp upgrade into the main fork since we'll want to do that anyway, but if you do have an opportunity to test on your other machine later that'd be fantastic -- I definitely appreciate the help.  And I totally agree on this being a particularly frustrating issue to track down and fix :)

**Comment by dctrud** [comment]
@dannon - big apologies for promising to test this further and then disappearing on you.... family emergency etc.

Do you still want me to test anything on the MacBook / vagrant environment I had regular errors with? I notice various others have been reporting the problem, so not sure where the testing stands.

Apologies again for going AWOL on this!

**Comment by dannon** [comment]
No problem at all.  The fixes I had lined up didn't resolve the issues, so I'm working on something new -- I can let you know when it's ready for hands-on testing though!

**Comment by dannon** [comment]
@dctrud If you're so inclined, I have a PR open that I think should finally put this to rest:  https://github.com/galaxyproject/galaxy/pull/198

You can pull the changes from the 'Kombuaaaargh" branch of my fork ( sense my frustration with this issue? ;) )

**Comment by jmchilton** [comment]
Need to build a planemo image - but I am going to mark this as solved by galaxyproject/galaxy#198 - which multiple people have reported as fixing the problem.
